src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,475 @@
+                    randomAlgorithm(numIdsToMake);
+                }
+                redundantIdList = DATABASE_MANAGER.checkId(ID_LIST);
+                    removeRedundantId(redundantIdList);
+
+                    // if no redundant ids were found, add new ids to database
+                    if (ID_LIST.size() == this.AMOUNT) {
+                        DATABASE_MANAGER.addId(ID_LIST);
+                    } else if (maxIdsToMake == redundantIdList.size()) {
+                        return "cannot make any more ids given parameters1";
+                    } else {
+                        // decreases the number of ids to make based on what was 
+                        // just added to ID_LIST
+                        numIdsToMake = this.AMOUNT - ID_LIST.size();
+                    }
+                } while (ID_LIST.size() < this.AMOUNT);
+            return convertListToJson();
+        } else {
+            return "cannot make any more ids given parameters";
+        }
+
+    }
+
+    /**
+     * Calculates the number of possible permutations in a given char mapping
+     * @return - number of permutations
+     */
+    private long calculateCharMapPermutations() {
+        long numPermutations = 1;
+        for (int i = 0; i < CHAR_MAP.length(); i++) {
+            /* when it isn't auto, we will not check the database as that
+             will not produce an accurate reseult. What will be returned 
+             instead is the comparison between the maximum number of 
+             permutations and the requested amount.
+             */
+
+            String charToken = BASE_MAP.get(CHAR_MAP.charAt(i));
+            System.out.println("charToken = " + charToken);
+            int base = tokenMaps.get(charToken).length();
+            System.out.println("base = " + base);
+            numPermutations *= base;
+            System.out.println("numPermutations = " + numPermutations);
+        }
+        return numPermutations;
+    }
+    
+    
+    /**
+     * Generates random ids based on char mappings. The ids are stored in
+     * ID_LIST
+     * @param numIdsToMake - number of ids to make     
+     */
+    private void randomAlgorithm(int numIdsToMake){          
+        String tempId, charToken;
+        String charRange;
+        int randomNum;
+                
+        while (ID_LIST.size() < numIdsToMake) {
+            tempId = PREFIX; 
+            for (int j = 0; j < CHAR_MAP.length(); j++) {
+                charToken = BASE_MAP.get(CHAR_MAP.charAt(j));
+                charRange = tokenMaps.get(charToken);
+                randomNum = (int)(Math.random() * charRange.length());
+                
+                tempId += charRange.charAt(randomNum);
+                                                
+            }
+            ID_LIST.add(tempId);
+        }
+                
+    }
+
+    /**     
+     * In the for loop, k represents last index of char mapping. This for 

```

Comment:
Method javadocs should be restricted to just high level descriptions of what the method does.  Things like implementation detials should be keep as normal comments